app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    if params[:question]
+      @question = Question.find_by(id: params[:question])
+    elsif params[:answer]
+      @answer = Answer.find_by(id: params[:answer])
+    end
+  end
+
+  def create
+    @question = Question.find_by(id: params[:comment][:question_id])
+    @answer = Answer.find_by(id: params[:comment][:answer_id])
+    @comment = Comment.new(comment_params)
+    if @question
+      if @comment.save
+        flash[:notice] = "Comment created"
+        redirect_to question_path(@question)
+      else
+        flash[:notice] = "Invalid Comment"
+        redirect_to question_path(@question)
+      end
+    elsif @answer
+      if @comment.save
+        flash[:notice] = "Comment created"
+        redirect_to question_path(@answer.question.id)
+      else
+        flash[:notice] = "Invalid Comment"
+        redirect_to question_path(@answer.question.id)
+      end
+    end
+  end
+
+  def destroy
+    @comment = Comment.find_by(id: params[:id])
+    if current_user == @comment.user
```

Comment:
These kinds of permissions might be better on the model - so you could have 
```
module Ownership
  def owned_by?(user)
    self.user == user
  end
end
```
Then mix it into question / answer / comment